src/table/use-column-widths.tsx

```diff
@@ -140,11 +148,24 @@ export function ColumnWidthsProvider({ visibleColumns, resizableColumns, contain
   });
 
   // Observes container size and requests an update to the last cell width as it depends on the container's width.
-  useResizeObserver(containerRef, ({ contentBoxWidth: containerWidth }) => {
+  useResizeObserver(measureRef, ({ contentBoxWidth: containerWidth }) => {
     containerWidthRef.current = containerWidth;
     updateColumnWidths();
   });
 
+  useMutationObserver(
+    containerRef,
+    () => {
```

Comment:
This can not only impact the rendering performance but also the runtime performance of features such as resizable columns and sticky columns, where the updates to styles are bound to the pointer movements.

Reply:
Good point. I wonder whether observing each cell individually might have some benefits in that sense. I'll continue investigating and experimenting

Reply:
Why the mutation observer is needed? Theoretically, any update to the table cell should cause browser to update its width accordingly which should cause the measure div to grow and trigger the resize observer.

Reply:
I don't think observing each cell is needed. Whenever the browser increases size of a single cell it also affect the size of the entire column and entire table. It should be enough to only observe the table (but not the container which is scrollable and does not increase its width).

Reply:
ah yes, got what you mean now about the measure div. I'll look into this. Thanks for the insight!